pyanaconda/modules/network/nm_client.py

```diff
@@ -51,6 +53,46 @@
 }
 
 
+@contextmanager
+def nm_client_in_thread():
+    """Create NM Client with new GMainContext to be run in thread.
+
+    For more information see NetworkManager example examples/python/gi/gmaincontext.py
+    """
+    mainctx = create_new_context()
+    mainctx.push_thread_default()
+
+    try:
+        yield get_new_nm_client()
+    finally:
+        mainctx.pop_thread_default()
+
+
+def get_new_nm_client():
+    """Get new instance of NMClient.
+
+    :returns: an instance of NetworkManager NMClient or None if system bus
+              is not available or NM is not running
+    :rtype: NM.NMClient
+    """
+    if not SystemBus.check_connection():
+        log.debug("get new NM Client failed: SystemBus connection check failed.")
+        return None
```

Comment:
Btw, this check seems unnecessary. `nm_client_new()` is documented to only fail for one reason (plus cancellation): if the call to `g_bus_get()` fails. And event that failure could be avoided by passing a `GDBusConnection` first.